### DISCUSSION TITLE
how to find a reason for undirected graph for Data(...) format in PyG

### DISCUSSION THREAD

**Discussion by Amadeus-System** [pyg-team/pytorch_geometric#4267] [Q&A]
Hi. I am a researcher in condensed matter physics.
I am preprocessing crystal graph dataset to Data format in PyG. 

My Dataset should be converted to **undirected, multi graph (which is possible to have multiple edges between nodes, but each edge between two nodes can have different edge_attr)**.

I have tested how Data() format in PyG work with simple graph example, after that I have created preprocessed custom dataset for my own.

However, there is some unknown reason. 

![capture1](https://user-images.githubusercontent.com/76824867/158521726-7ad20ad4-987a-498b-a1b7-da70b892937c.PNG)

as you can see in the above figure, each Data() format represents undirected, multi edge graph object. but when I use 'is_undirected()' method to check whether the graph is undirected, it returns false.
Since each Data() represents a single undirected-multi graph, it has repeated edge_index and repeated edge_attr. 
And indexes of edge_index and edge_attr are corresponed, of course. 

I have checked edge_index and edge_attr. but I cannot see any problems.
could you tell me how I can check the reason which make my dataset **directed** ??

**Comment by rusty1s** [comment]
Thanks for reporting. Our `is_undirected` functionality can be found here: https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/utils/undirected.py#L11-L51

Can you copy-paste the functionality and see where it might break on your end? Alternatively, you can provide a simple example in which `is_undirected` breaks for you. It might be related to using `torch.all` rather than `torch.allclose`.
